### PR TITLE
Port handlers/ to Python 3

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

File: changelog.d/3803.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+storage/ is now ported to Python 3.
```
Comment (erikjohnston) [inline changelog.d/3803.misc]: handlers/

File: synapse/handlers/auth.py [hunk 2]
```diff
@@ -895,22 +895,24 @@ def validate_hash(self, password, stored_hash):
...
 
             return bcrypt.checkpw(
                 pw.encode('utf8') + self.hs.config.password_pepper.encode("utf8"),
-                stored_hash.encode('utf8')
+                stored_hash
             )
 
         if stored_hash:
+            if not isinstance(stored_hash, bytes):
+                stored_hash = stored_hash.encode('ascii')
```
Comment (erikjohnston) [inline synapse/handlers/auth.py]: I assume changing from using 'utf8' to 'ascii' is fine?

File: synapse/handlers/e2e_keys.py [hunk 3]
```diff
@@ -330,7 +330,8 @@ def _upload_one_time_keys_for_user(self, user_id, device_id, time_now,
                         (algorithm, key_id, ex_json, key)
                     )
             else:
-                new_keys.append((algorithm, key_id, encode_canonical_json(key)))
+                new_keys.append((
+                    algorithm, key_id, encode_canonical_json(key).decode('ascii')))
```
Comment (erikjohnston) [inline synapse/handlers/e2e_keys.py]: Nit picky, but this is quite unreadable to me at the end, can you stick the trailing `))` on a newline?

File: synapse/handlers/e2e_keys.py [hunk 4]
```diff
@@ -358,7 +359,7 @@ def _exception_to_failure(e):
     # Note that some Exceptions (notably twisted's ResponseFailed etc) don't
     # give a string for e.message, which json then fails to serialize.
     return {
-        "status": 503, "message": str(e.message),
+        "status": 503, "message": str(e.args[0]),
```
Comment (erikjohnston) [inline synapse/handlers/e2e_keys.py]: don't exceptions have a `message`?

File: synapse/handlers/auth.py (hunk 2 above)
Comment (hawkowl) [reply synapse/handlers/auth.py]: The hashes are ASCII hex, so yes.

File: synapse/handlers/e2e_keys.py (hunk 4 above)
Comment (hawkowl) [reply synapse/handlers/e2e_keys.py]: not on python 3, `.message` is removed

File: synapse/handlers/e2e_keys.py (hunk 3 above)
Comment (hawkowl) [reply synapse/handlers/e2e_keys.py]: can we save that for running black on things? :)

File: changelog.d/3803.misc (hunk 1 above)
Comment (hawkowl) [reply changelog.d/3803.misc]: whoops

File: synapse/handlers/e2e_keys.py (hunk 3 above)
Comment (erikjohnston) [reply synapse/handlers/e2e_keys.py]: Sure, so long as we do do it

File: synapse/handlers/e2e_keys.py (hunk 4 above)
Comment (erikjohnston) [reply synapse/handlers/e2e_keys.py]: Can we just do `str(e)` tbh, otherwise it feels a bit too awful

**Pull request merged**